packages/core/database/lib/entity-manager.js

```diff
@@ -402,6 +402,21 @@ const createEntityManager = (db) => {
             continue;
           }
 
+          // delete previous relations
+          const where = { $or: [] };
+
+          for (const { field, related_type: relatedType, related_id: relatedId } of rows) {
+            const isMorphOne =
+              db.metadata.get(relatedType).attributes[field].relation === 'morphOne';
+            if (isMorphOne) {
+              where.$or.push({ related_type: relatedType, field, related_id: relatedId });
```

Comment:
you could simplify by pushing those conditions in one simple array and applying the $or in the .where({ $or: conditions })) call